### DISCUSSION TITLE
docker mastodon 4.3.1  all files are owned by root. Can't change max_chars

### DISCUSSION THREAD

**Discussion by GerritKuilder** [mastodon/mastodon#32663] [Q&A] [answered]
Hi All,

I updated my docker compose file to the 4.3.1 images. I want to update the MAX_CHARS from 500 to 5000 

I use the commands that used to work:
sudo docker exec live-web-1 sed -i 's/500/5000/g' app/validators/status_length_validator.rb

Result: sed: couldn't open temporary file app/validators/sedwbDhFq: Permission denied

I checked /opt/mastodon and noticed that they are all owned by root:
drwxr-xr-x 16 root     root       4096 Oct 21 09:01 app

1) Anyone seen this and/or have a solution for this?
2) is there another image I can pull that would allow easier modification of the MAX_CHARS?

Any help would be appreciated.

Kind Regards,

Gerrit Kuilder

**Comment by GerritKuilder** [comment]
I looked at glitch-soc but I don't want to go through 
I am tempted to do a test with these images, anyone any experience with this?
https://docs.linuxserver.io/images/docker-mastodon/  as this has a tag for glitch.

My instance has been running happily for quite a while and I was able to set the max_chars to a larger number before upgrading to 4.3.1.

The main issue seems to come from all the files being owned by root. 
I also see an issue with images not being preloaded for certain posts, for exmple: https://mastodon.ozioso.online/deck/@RTENewsHeadlinesRSS/113377864617759168 ( all posts with a URL to webpages and a preview image. This might be a permission issue, as they show fine on other mastodon instances.

KR, Gerrit

**Reply by GerritKuilder** [reply]
One part is fixed:
>I also see an issue with images not being preloaded for certain posts, for exmple: >https://mastodon.ozioso.online/deck/@RTENewsHeadlinesRSS/113377864617759168 ( all posts with a URL to webpages and a >preview image. This might be a permission issue, as they show fine on other mastodon instances.

A change/difference of the public directory between my original and new docker compose. So that's one less thing to deal with.

KR, Gerrit

**Comment by neuhaus** [comment, marked as answer]
You can use `docker exec -u 0` to run something as root in the container.

**Reply by GerritKuilder** [reply]
Haven't tested that as I moved to glitch-soc as this was easier to configure.
A good tip anyway, thank you.

Kind Regards,

Gerrit

**Reply by neuhaus** [reply]
For the record, here's what I use in my docker compose file to start mastodon-web. Because the sed invocation is idempotent, it doesn't hurt to do it on every restart. The only tricky part is that sed has to run as root whereas the "bundle" runs as mastodon:

```
    user: "0"
    command: >
      sh -c "
        sed -i 's/MAX_CHARS = 500/MAX_CHARS = 1500/' /opt/mastodon/app/validators/status_length_validator.rb
        rm -f /mastodon/tmp/pids/server.pid
        exec su mastodon -c 'bundle exec rails s -p 3000'
      "
```

**Reply by neuhaus** [reply]
In the same fashion, you can patch the maximum number of poll options and their lengths:

`sed -i 's/MAX_OPTIONS      = 4/MAX_OPTIONS      = 10/;s/MAX_OPTION_CHARS = 50/MAX_OPTION_CHARS = 100/' /opt/mastodon/app/validators/poll_options_validator.rb`
